### PR TITLE
add distributed lock service api (WP-643)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: src/DistributedLockService/DistributedLockServiceApi.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    /**
+     * @param string $key
+     * @return void
+     * @throws SmartlingApiException
+     */
+    public function releaseLock($key)
+    {
+        $this->sendRequest("locks/$key", $this->getDefaultRequestData('json', []), self::HTTP_METHOD_DELETE);
```
Comment (PavelLoparev) [inline src/DistributedLockService/DistributedLockServiceApi.php:55]: return? Sdk handles `{data: null}` responses and returns bool.

**Edited**: I agree that there is no much sense in always returning `true` if it's not failed. Ignore this comment

**Review by PavelLoparev — CHANGES_REQUESTED** [review]
Also, please add unit tests. Take a look at any other api client tests

File: src/DistributedLockService/DistributedLockServiceApi.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return $instance;
+    }
+
+    /**
+     * @param string $key
+     * @param float|int $ttlSeconds
+     * @return \DateTime lock release time
+     * @throws SmartlingApiException
+     */
+    public function acquireLock($key, $ttlSeconds)
```
Comment (PavelLoparev) [inline src/DistributedLockService/DistributedLockServiceApi.php]: There are also `timeout` and `wait` optional parameters.

**Review by dimitrystd — APPROVED** [review]

File: src/DistributedLockService/DistributedLockServiceApi.php (hunk 2 above)
Comment (vsolovei-smartling) [reply src/DistributedLockService/DistributedLockServiceApi.php]: Done

**Pull request merged**